src/inlineSuggestions/stateTracker.ts

```diff
@@ -34,3 +43,21 @@ export function initTracker(): Disposable[] {
     window.onDidChangeTextEditorSelection(onTextSelectionChange),
   ];
 }
+
+class Change {
+  constructor(private readonly contentChange: TextDocumentContentChangeEvent) {}
+
+  isValid(): boolean {
+    return (
+      this.contentChange?.rangeLength >= 0 && this.contentChange?.text !== ""
+    );
+  }
+
+  isSingleCharChange() {
+    return this.contentChange?.text.trim().length <= 1;
+  }
+
+  isNotIndentationChange() {
+    return this.contentChange?.text !== " ".repeat(getTabSize());
+  }
+}
```

Comment:
rename to something more specific (`TabnineDocumentChange` for example) and extract to separate file